test/e2e/e2e_test.go

```diff
@@ -0,0 +1,147 @@
+package e2e
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/xmudrii/etcdproxy-controller/pkg/apis/etcd/v1alpha1"
+	clientset "github.com/xmudrii/etcdproxy-controller/pkg/client/clientset/versioned"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// kubeconfigPath is path to the kubernetes configuration file.
+var kubeconfigPath = filepath.Join(os.Getenv("HOME"), "/.kube/config")
+
+// newKubeClient create new Kubernetes client instance based on kubeconfig file.
+func newKubeClient(kubeconfigPath string) (*kubernetes.Clientset, error) {
+	// use the current context in kubeconfig.
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// create the clientset.
+	return kubernetes.NewForConfig(config)
+}
+
+func newEtcdProxyClinet(kubeconfigPath string) (*clientset.Clientset, error) {
+	// use the current context in kubeconfig.
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// create the clientset.
+	return clientset.NewForConfig(config)
+}
+
+func replicasetName(etcdstorageName string) string {
+	return fmt.Sprintf("etcd-rs-%s", etcdstorageName)
+}
+
+func serviceName(etcdstorageName string) string {
+	return fmt.Sprintf("etcd-%s", etcdstorageName)
+}
+
+func TestDeployEtcdStorage(t *testing.T) {
```

Comment:
Is it okay to have all these checks in this function, or I should split them to multiple tests? In this function we're testing is the EtcdStorage instance deployed, and are all relevant resources deployed.

TODO: add godoc comment

Reply:
Added godoc comment.